support-frontend/assets/pages/[countryGroupId]/checkout.tsx

```diff
@@ -803,23 +824,42 @@ function CheckoutComponent({ geoId }: Props) {
 										{showStateSelect && (
 											<StateSelect
 												countryId={countryId}
-												state={billingState}
+												state={billingState ?? ''}
 												onStateChange={setBillingState}
-												error={undefined}
+												error={billingStateError}
 											/>
 										)}
 
 										{countryId === 'US' && (
 											<div>
 												<TextInput
 													id="zipCode"
-													name="billing-postcode"
 													label="ZIP code"
-													value={billingPostcode}
-													error={undefined}
+													name="billing-postcode"
 													onChange={(event) =>
 														setBillingPostcode(event.target.value)
 													}
+													maxLength={20}
+													value={billingPostcode}
+													pattern={doesNotContainEmojiPattern}
```

Comment:
Should we be using this?
https://github.com/guardian/support-frontend/blob/main/support-frontend/assets/helpers/forms/formValidation.ts#L37-L38

Reply:
I dont think this is used? Its a full house format I think (https://stackoverflow.com/questions/2577236/regex-for-zip-code) so `12345-1234` rather than `AL1234` for an area (I think). Dont think much fomatting other than 20chars is done on production?

Reply:
OK! Sounds good.